src/Microsoft.ML.TimeSeries/ExtensionsCatalog.cs

```diff
@@ -143,9 +146,59 @@ public static class TimeSeriesCatalog
         /// </format>
         /// </example>
         public static SrCnnAnomalyEstimator DetectAnomalyBySrCnn(this TransformsCatalog catalog, string outputColumnName, string inputColumnName,
-            int windowSize=64, int backAddWindowSize=5, int lookaheadWindowSize=5, int averageingWindowSize=3, int judgementWindowSize=21, double threshold=0.3)
+            int windowSize = 64, int backAddWindowSize = 5, int lookaheadWindowSize = 5, int averageingWindowSize = 3, int judgementWindowSize = 21, double threshold = 0.3)
             => new SrCnnAnomalyEstimator(CatalogUtils.GetEnvironment(catalog), outputColumnName, windowSize, backAddWindowSize, lookaheadWindowSize, averageingWindowSize, judgementWindowSize, threshold, inputColumnName);
 
+        /// <summary>
+        /// Create <see cref="RootCause"/>, which localizes root causes using decision tree algorithm.
+        /// </summary>
+        /// <param name="catalog">The anomaly detection catalog.</param>
+        /// <param name="src">Root cause's input. The data is an instance of <see cref="Microsoft.ML.TimeSeries.RootCauseLocalizationInput"/>.</param>
+        /// <param name="beta">Beta is a weight parameter for user to choose. It is used when score is calculated for each root cause item. The range of beta should be in [0,1]. For a larger beta, root cause point which has a large difference between value and expected value will get a high score. On the contrary, for a small beta, root cause items which has a high relative change will get a high score.</param>
+        /// <example>
+        /// <format type="text/markdown">
+        /// <![CDATA[
+        /// [!code-csharp[LocalizeRootCause](~/../docs/samples/docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/TimeSeries/LocalizeRootCause.cs)]
+        /// ]]>
+        /// </format>
+        /// </example>
+        public static RootCause LocalizeRootCause(this AnomalyDetectionCatalog catalog, RootCauseLocalizationInput src, double beta = 0.5)
+        {
+            //check the root cause input
+            CheckRootCauseInput(src);
+
+            //check beta
+            if (beta < 0 || beta > 1) {
+                throw new ArgumentException("Beta must be in [0,1]");
```

Comment:
Can you please follow the examples of other places in code where parameter validation is done? You can get a reference to IHostEnvironment using CatalogUtils.GetEnvironment(catalog) and then  after a call to Register to get IHost, you can use host.CheckXXX functions for parameter validation.

Reply:
> Can you please follow the examples of other places in code where parameter validation is done? You can get a reference to IHostEnvironment using CatalogUtils.GetEnvironment(catalog) and then after a call to Register to get IHost, you can use host.CheckXXX functions for parameter validation.

updated